### PR TITLE
Fix minor grammar issues in the readme

### PR DESCRIPTION
**Pull request by evg-tso**
Changes:
- Apply [grammarly](http://grammarly.com) suggestions to the README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ changed, then deliver the new handler to the promise which unblocks
 the server.
 
 Note that we only need to go to this additional effort if retaining
-open resources is useful during development, otherwise we can rely on
+open resources is useful during development, otherwise, we can rely on
```
Comment (weavejester) [inline README.md:230]: I think the comma you've placed here isn't correct.

**Review by weavejester — CHANGES_REQUESTED** [review]
Thanks for the review! I think in a few places the existing grammar is correct, but most of the updates are correct.

File: README.md [hunk 2]
```diff
@@ -86,7 +86,7 @@ And load it with Integrant's version of `read-string`:
 ### Initializing and halting
 
 Once you have a configuration, Integrant needs to be told how to
-implement it. The `init-key` multimethod takes two arguments, a key
+implement it. The `init-key` multimethod takes two arguments, a key,
```
[the remark this answers is not in the document]
Comment (evg-tso) [reply README.md:89]: The suggestion is baed on the [Oxford comma](https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/).  
I agree it's a gray area and can be removed. WDYT?

[the remark this answers is not in the document]
Comment (weavejester) [reply README.md:89]: In this case I believe this is a false positive from Grammarly, as this isn't an Oxford comma.

[the remark this answers is not in the document]
Comment (evg-tso) [reply README.md]: Removed it

File: README.md (hunk 1 above)
Comment (evg-tso) [reply README.md]: Removed it

File: README.md [hunk 3]
```diff
@@ -104,12 +104,12 @@ Keys are initialized recursively, with the values in the map being
...
 
 In the configuration we defined before, `:handler/greet` will be
-initialized first, and it's value replaced with a handler function.
+initialized first, and its value replaced with a handler function.
 When `:adapter/jetty` references `:handler/greet`, it will receive the
 initialized handler function, rather than the raw configuration.
 
 The `halt-key!` multimethod tells Integrant how to stop and clean up
-after a key. Like `init-key`, it takes two arguments, a key and its
+after a key. Like `init-key`, it takes two arguments, a key, and its
```
[the remark this answers is not in the document]
Comment (evg-tso) [reply README.md]: Removed it

**Comment by weavejester** [comment]
Thanks. Can you squash your commits and change the message to something like

    Fix grammatical errors in README

    Corrections come from Grammarly.

**Comment by evg-tso** [comment]
Done.

**Pull request merged**